### PR TITLE
Swo 214 error handler middleware

### PR DESCRIPTION
**Pull request by shinaBR2**
## Summary by CodeRabbit

* **New Features**
  * Centralized error handler that returns clear user-facing messages, explicit retry hints, preserves context, and supports optional error-tracking integration (PostHog) when configured.
  * Errors now expose a user-facing message and a notification flag to control reporting.

* **Tests**
  * Expanded test suite covering error paths, notification behavior, tracking integration, logging, and edge cases.

* **Chores**
  * Added dependencies and a TypeScript path alias to support the above.

### CODE REVIEW

File: src/middleware/errorHandler/index.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+  /**
+   * TODO
+   * Calcuate from context
+   * - Hasura Action: always 400
+   * - Hasura Event Trigger: 400 or 200 based on shouldRetry
+   * - Stripe Webhook: 400 or 200 based on shouldRetry
+   * - Cloud Tasks: 400 or 200 based on shouldRetry
+   */
+  const statusCode = 400;
```
[the remark this answers is not in the document]
Comment (shinaBR2) [reply src/middleware/errorHandler/index.ts]: Nice catch, I will fix this in later PR

[the remark this answers is not in the document]
Comment (shinaBR2) [reply src/middleware/errorHandler/index.ts]: yes please

File: src/middleware/errorHandler/index.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+          host: errorTracker.posthogHost,
+        });
+        posthog.captureException(error, userId);
+        // https://posthog.com/docs/error-tracking/installation/hono
+        posthog.shutdown();
+      } catch (error) {
+        // This will trigger GCP Error Reporting
+        logger.error(error, 'Failed to send error to PostHog');
+      }
+    }
```
[the remark this answers is not in the document]
Comment (shinaBR2) [reply src/middleware/errorHandler/index.ts]: Yes please

File: src/middleware/errorHandler/index.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+          hasuraAction,
+          hasuraEventTrigger,
+        },
+      });
+
+      return {
+        result: error.toUserResponse(),
+        statusCode,
+      };
+    }
```
[the remark this answers is not in the document]
Comment (shinaBR2) [reply src/middleware/errorHandler/index.ts]: No this is my intention, please learn it. Notify mean send alert to dev team, we achieve it by log errors and GCP error reporting will do that job. When notify is false, that means errors like validation errors, we should treat it as analytics.

File: src/middleware/errorHandler/index.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        // error: "Internal Server Error",
+        message: error.message,
+        extensions: {
+          code: ERROR_CODES.UNEXPECTED_ERROR,
+          shouldRetry: true,
+        },
+      },
+      statusCode,
+    };
+  }
```
[the remark this answers is not in the document]
Comment (shinaBR2) [reply src/middleware/errorHandler/index.ts]: Same explain above

File: src/utils/custom-error/index.ts [hunk 5]
```diff
@@ -99,9 +108,16 @@ class CustomError extends Error {
     this.timestamp = Date.now();
     this.errorCode = errorCode;
     this.severity = severity;
-    this.shouldRetry = shouldRetry;
     this.contexts = [];
 
+    const errorConfig = ERROR_CONFIG[errorCode as keyof typeof ERROR_CONFIG];
```
[the remark this answers is not in the document]
Comment (shinaBR2) [reply src/utils/custom-error/index.ts]: We already handled the case `errorConfig` is `undefined` at the next lines
